src/routines.ts

```diff
@@ -70,3 +75,31 @@ export const subscribeRoutine = (
       })
     )
     .subscribe();
+
+type PlainOrArray<T> = T | T[];
+export const ensureArray = <T>(a: PlainOrArray<T>): T[] =>
+  Array.isArray(a) ? a : [a];
+
+type TapRoutine = {
+  <Payload>(
+    actionCreator: PlainOrArray<UnknownActionCreatorWithPayload<Payload>>,
+    body: (payload: Payload) => void
+  ): Routine<void>;
+
+  (
+    actionCreator: PlainOrArray<UnknownActionCreator>,
+    body: () => void
+  ): Routine<void>;
+};
+
+export const tapRoutine: TapRoutine = <Payload>(
+  actionCreator: PlainOrArray<
+    UnknownActionCreatorWithPayload<Payload> | UnknownActionCreator
+  >,
+  body: (payload?: Payload) => void
+) =>
+  routine(
+    ofType(...(ensureArray(actionCreator) as any)),
```

Comment:
```suggestion
    ofType(
      ...(ensureArray(
        actionCreator
      ) as UnknownActionCreatorWithPayload<Payload>[])
    ),
```

this isn't great, but I think the `as` is a little better